### DISCUSSION TITLE
Development time

### DISCUSSION THREAD

**Discussion by jmalek-hexagon** [hotosm/tasking-manager#4558] [General]
Hi,
as a developer who would like to start a project of roughly similar size and type I am really curious about how many man-days this project consumed to date.
Your answer would be greatly appreciated as it would provide me and the other developers with roughly how much time we have to calculate in the scope of the following year(s) or to decide whether to dump the idea all together.
Thanks!
Jan

**Comment by willemarcel** [comment]
@jmalek-hexagon It's quite difficult to answer your question, because TM4 was a redesign of TM3 and what we have now is a result of many years of work. Also, we had people dividing time between TM and other projects and each one committing a different number of hours.

I'm not good on estimation, but I would bet you can build something similar with a part-time designer and 5 full time senior developers, during 12 months. I subscribed TM to Open Hub, soon you can check their cost estimation: https://www.openhub.net/p/tasking-manager

**Reply by jmalek-hexagon** [reply]
@willemarcel Thank you very much for your quick response! This is very useful us as we already spent weeks of designing and prototyping components of the application. Its's driving me crazy that still at this point it seems that everyone else thinks it can by done in a week by any two developers (this has already gone up from one developer and a weekend). So if you won't mind I will present this project and your estimation as a reference of what our project might take to hopefully open the eyes of my colleagues.

**Reply by willemarcel** [reply]
Good to know it will be helpful. I believe the era of software made by two developers in a week is definitely gone. Nowadays, everyone expects that your software provides the same level of experience they have on the tech giants' products. So the bar is much higher than some years ago, and it's difficult to build products that matches with that expectation.
